### DISCUSSION TITLE
Flow readning data from existing collection items and compare with external api json data then updates the differences

### DISCUSSION THREAD

**Discussion by c001os** [directus/directus#20929] [Feature Requests]
### Summary

Hi! I want to create a flow (or multiple flows) which can achieve the following:
- Read all items data from existing directus collection
- Get external api api json data and maps correctly like the existing directus collection (already done)
- Compare the directus collection data with external api data
- Update directus collection data if there is differences

Any idea how or can it be achieved?

Many many thanx

### Motivation

Make connection between our web app based on directus and our MS Business Central data.

### Detailed Design

- Read all items data from existing directus collection
- Get external api api json data and maps correctly like the existing directus collection (already done)
- Compare the directus collection data with external api data
- Update directus collection data if there is differences

### Must Have:
- Comaprinson and data update

### Should Have:
- One flow to achieve in multiple items

### Drawbacks

-

### Alternatives

-

### Adoption Strategy

-

**Comment by br41nslug** [comment]
All the individual items in "detailed design" sound like things Flows are already capable of currently 🤔 What exactly is the feature you are requesting here?

> Read all items data from existing directus collection

This is what the "Read Data" operation does / can do.

> Get external api api json data and maps correctly like the existing directus collection (already done)

For this there is the "Request URL" operation to handle external request

> Compare the directus collection data with external api data

For this one either the "Run Script" operation or a custom operation is probably most appropriate (depending on the complexity)

> Update directus collection data if there is differences

When worked out what to update in the last step you can use the "Update Data" operation to update.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
